assets/js/components/ChecksSelection/ChecksSelectionHeader.test.jsx

```diff
@@ -0,0 +1,142 @@
+
+import userEvent from '@testing-library/user-event';
+import ChecksSelectionHeader from './ChecksSelectionHeader';
+import { renderWithRouter } from '../../lib/test-utils';
+
+describe('ChecksSelectionHeader component', () => {
+  it('should render a target checks selection header', async () => {
+    const user = userEvent.setup();
+
+    const targetID = faker.datatype.uuid();
+    const targetName = faker.lorem.word();
+    const selection = [faker.datatype.uuid(), faker.datatype.uuid()];
+    const savedSelection = [faker.datatype.uuid()];
+    const onSaveSelection = jest.fn();
+    const onStartExecution = jest.fn();
+
+    renderWithRouter(
+      <ChecksSelectionHeader
+        targetID={targetID}
+        targetName={targetName}
+        backTo={<button type="button">Back to Target Details</button>}
+        pageHeader={<div>Target Check Settings</div>}
+        isSavingSelection={false}
+        selection={selection}
+        savedSelection={savedSelection}
+        onSaveSelection={onSaveSelection}
+        onStartExecution={onStartExecution}
+      />
+    );
+
+    expect(screen.getByText('Target Check Settings')).toBeVisible();
+    expect(
+      screen.getByRole('button', { name: 'Back to Target Details' })
+    ).toBeVisible();
+    expect(
+      screen.getByRole('button', { name: 'Save Checks Selection' })
+    ).toBeVisible();
+    expect(
+      screen.getByRole('button', { name: 'Start Execution' })
+    ).toBeVisible();
+    expect(
+      screen.queryByText(
+        'Click Start Execution or wait for Trento to periodically run checks.'
+      )
+    ).toBeVisible();
+
+    // Saving a selection
+    await user.click(screen.getByText('Save Checks Selection'));
+
+    expect(onSaveSelection).toHaveBeenCalledWith(
+      selection,
+      targetID,
+      targetName
+    );
+
+    // Starting an execution
+    await user.click(screen.getByText('Start Execution'));
+
+    expect(onStartExecution).toHaveBeenCalled();
+  });
+
+  it('should not allow saving a selection', async () => {
+    const user = userEvent.setup();
+
+    const targetID = faker.datatype.uuid();
+    const targetName = faker.lorem.word();
+    const selection = [faker.datatype.uuid(), faker.datatype.uuid()];
+    const onSaveSelection = jest.fn();
+
+    renderWithRouter(
+      <ChecksSelectionHeader
+        targetID={targetID}
+        targetName={targetName}
+        backTo={<button type="button">Back to Target Details</button>}
+        pageHeader={<div>Target Check Settings</div>}
+        isSavingSelection
+        selection={selection}
+        savedSelection={selection}
+        onSaveSelection={onSaveSelection}
+        onStartExecution={() => {}}
+      />
+    );
+
+    expect(screen.getByText('Save Checks Selection')).toBeDisabled();
+
+    await user.click(screen.getByText('Save Checks Selection'));
+
+    expect(onSaveSelection).not.toHaveBeenCalled();
+  });
+
+  const executionDisallowedScenarios = [
```

Comment:
When possible, I tend to prefer straight words instead of negations of others, sounds clearer and more explicit. Can we live with  `Disallowed` instead of `NotAllowed`? :smile:

Reply:
Alright, no problemo